test_unit.py

```diff
@@ -1317,11 +1415,12 @@ def test_keyreg_txn_online(self):
 
     def test_keyreg_txn_offline(self):
         keyregtxn = (
-            "hqNmZWXNA+iiZnbOALutq6JnaMQgSGO1GKSzyE7IEPItTxCByw9x8FmnrCDexi9/c"
-            "OUJOiKibHbOALuxk6NzbmTEIAn70nYsCPhsWua/bdenqQHeZnXXUOB+jFx2mGR9tu"
-            "H9pHR5cGWma2V5cmVn")
+            "h6NmZWXNA+iiZnbOALutq6JnaMQgSGO1GKSzyE7IEPItTxCByw9x8FmnrCDexi9/c"
```

Comment:
This change was necessary to add a new `nonpart` field into the message as `KeyregTxn` undictify sorts it into the respective online or offline transaction.

Reply:
I think the change to this test should be reverted, since now we have online, offline, and nonparticipating keyreg txns, so the original implementation is correct again.

Reply:
This has been reverted and the test passes, but I had to make a change to another part of the code which I will tag you on and explain it.